### DISCUSSION TITLE
zppy issue on NERSC: No architecture specified

### DISCUSSION THREAD

**Discussion by BunnyVon** [E3SM-Project/zppy#416] [Questions]
I received an error after I ran zppy. 
`b'sbatch: error: No architecture specified, cannot estimate job costs.\nsbatch: error: Batch job submission failed: Unspecified error\n'`
it turned out that a line `#SBATCH -C haswell` will need to add in `climo_atm_monthly_180x360_aave_*.bash` to make it run.  Does anyone know how to fix in `zppy` code?  Thanks.

**Comment by forsyth2** [comment]
`haswell` is the `constraint` -- https://github.com/E3SM-Project/zppy/blob/main/zppy/templates/slurm_header.sh#L19, https://github.com/E3SM-Project/zppy/blob/main/zppy/templates/default.ini#L14

**Reply by BunnyVon** [reply]
seems not working
```
[default]
environment_commands = "source /global/common/software/e3sm/anaconda_envs/load_latest_e3sm_unified_cori-haswell.sh"
input = /global/cfs/cdirs/e3sm/feng809/E3SM_output/lnd-atm_control
output = /global/cfs/cdirs/e3sm/feng809/paper01/data/lnd-atm_control
case = CBGCv1.LR.BGC-LNDATM.20TR
constraint = "haswell"
www = /global/cfs/cdirs/e3sm/www/sfeng/zppy_ouput
partition = debug
qos = debug
walltime = "00:30:00"
```
also, even though I set the `walltime` to be 30 min, it became 10 min after the submission. any clues?

**Reply by forsyth2** [reply]
I think `partition` should be `""`. If that doesn't work, perhaps `cori-haswell`. (I haven't done much on Cori lately, since it will be out of service soon. The Perlmutter example is https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/test_complete_run_pm-cpu.cfg).

As for the time shortening, I'm not sure what would cause that, but I do feel like we've seen that issue pop up a few times before.
